airflow/models/dag.py

```diff
@@ -3555,16 +3555,26 @@ def set_is_paused(self, is_paused: bool, including_subdags: bool = True, session
     def deactivate_deleted_dags(
         cls,
         alive_dag_filelocs: Container[str],
+        processor_subdir: str | None = None,
```

Comment:
Can we just make this a required argument? From what I can tell this is not used anywhere else, and although this is not explicitly marked as internal, I don’t think there’s reasonable use case to interact with DagModel directly.

Reply:
Same goes for DagCode I guess.

Reply:
Good idea. I double checked but indeed this is almost never used, only here and in 1 test that I adjusted.

We can even narrow the type to just `str` as `get_dag_directory` guarantee a str value as input.

Done :)